### ISSUE TITLE
Parse catch unknown in typescript 4.0

### ISSUE THREAD

**Issue by brendongo** [semgrep/semgrep#2613]
**Describe the bug**
In Typescript 4.0 the following is valid:

```
try {
    // ...
}
catch (e: unknown) {
    // error!
    // Property 'toUpperCase' does not exist on type 'unknown'.
    console.log(e.toUpperCase());

    if (typeof e === "string") {
        // works!
        // We've narrowed 'e' down to the type 'string'.
        console.log(e.toUpperCase());
    }
}
```

As of 0.40.0 semgrep fails to parse this as valid typescript

See: https://devblogs.microsoft.com/typescript/announcing-typescript-4-0/#unknown-on-catch

**To Reproduce**
For convenience: https://semgrep.dev/s/AvN2/

**Expected behavior**
Parses cleanly

**What is the priority of the bug to you?**
Is this a P0 (blocking your adoption of Semgrep or workflow), P1 (important to fix or quite annoying), P2 (regular bug that should get fixed)?
P2

**Comment by brendongo** [comment]
Once this passes can fully parse Zulip repo without excludes again

**Comment by mjambon** [comment]
Thanks @brendongo. It looks like `catch (e: whatever) {` isn't supported by tree-sitter-typescript. It should be easy to add to the grammar. I assigned it to me because at least I need to confirm and make a proper bug report or just fix it.

Oh, and our typescript setup in ocaml-tree-sitter is a little complicated so that's why it's probably best to assign the whole semgrep issue to me. However, someone else could do the fix in the grammar without difficulties. The changes in the ocaml code in semgrep-core should be minor to moderate, depending on how many other earlier changes this brings in.

**Comment by mjambon** [comment]
Tracked at https://github.com/tree-sitter/tree-sitter-typescript/issues/142

**Comment by mjambon** [comment]
The fix will be in semgrep 0.56 due this week.
